### PR TITLE
fix(userstatus): run user status automation job for start and end events 

### LINKED ISSUE #47058: [Bug]: User status automation for OOO is not triggering / resetting correctly
### ⚠️ This issue respects the following points: ⚠️

- [X] This is a **bug**, not a question or a configuration/webserver/proxy issue.
- [X] This issue is **not** already reported on [Github](https://github.com/nextcloud/server/issues?q=is%3Aopen+is%3Aissue+label%3Abug) OR [Nextcloud Community Forum](https://help.nextcloud.com/) _(I've searched it)_.
- [X] Nextcloud Server **is** up to date. See [Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) for supported versions.
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Bug description

When setting an Out of Office period, a custom status seems to block the Out of Office period, even when unsetting the custom status.

Additionally, Out of Office periods are not properly reset when they end.

### Steps to reproduce

1. Have custom status
2. Set Out of Office period
3. ???

### Expected behavior

The Out of office period takes precedence over a custom status.

### Installation method

None

### Nextcloud Server version

master

### Operating system

None

### PHP engine version

None

### Web server

None

### Database engine version

None

### Is this bug present after an update or on a fresh install?

None

### Are you using the Nextcloud Server Encryption module?

None

### What user-backends are you using?

- [ ] Default user-backend _(database)_
- [X] LDAP/ Active Directory
- [ ] SSO - SAML
- [ ] Other

### Configuration report

```shell
c.nc
```

### List of activated Apps

```shell
c.nc
```

### Nextcloud Signing status

```shell
c.nc
```

### Nextcloud Logs

```shell
c.nc
```

### Additional info

_No response_

### PR DESCRIPTION
**Pull request by miaulalala**
* Resolves: https://github.com/nextcloud/server/issues/47058

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [x] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [x] Screenshots before/after for front-end changes
- [x] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [x] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

**Review by st3iny — APPROVED** [review]
LGTM, as discussed yesterday.

File: apps/user_status/lib/Listener/OutOfOfficeStatusListener.php [hunk 1]
```diff
@@ -42,7 +44,10 @@
 		}
 
 		if ($event instanceof OutOfOfficeScheduledEvent
-			|| $event instanceof OutOfOfficeChangedEvent) {
+			|| $event instanceof OutOfOfficeChangedEvent
+			|| $event instanceof OutOfOfficeStartedEvent
+			|| $event instanceof OutOfOfficeEndedEvent
```
[the remark this answers is not in the document]
Comment (st3iny) [reply apps/user_status/lib/Listener/OutOfOfficeStatusListener.php:49]: Yeah, but it's better to explicit in this case. Dismissed.

**Review by ChristophWurst — APPROVED** [review]

**Pull request merged**